packages/spatial/src/renderer/csm/CSM.ts

```diff
@@ -463,26 +459,23 @@ export class CSM {
     }
   }
 
-  getExtendedBreaks(target: Vector2[]): void {
-    while (target.length < this.breaks.length) {
-      target.push(new Vector2())
-    }
-
-    target.length = this.breaks.length
+  getExtendedBreaks(): Vector2[] {
+    const target: Vector2[] = []
 
     for (let i = 0; i < this.cascades; i++) {
-      const amount = this.breaks[i]
+      const amount = this.breaks[i] || 0
       const prev = this.breaks[i - 1] || 0
-      target[i].x = prev
-      target[i].y = amount
+      target.push(new Vector2(prev, amount))
```

Comment:
why are we creating the vector here now? previously it was done lazily to avoid creaing garbage each frame

Reply:
The only thing I could find to stop the Three errors when resizing the canvas was changing it so each shader CSM uniform didn't reference the same array, having them reuse their unique arrays works too. I don't know why that fixed it though, if you have any ideas let me know